workflow/rules/trim.smk

```diff
@@ -6,22 +6,112 @@ rule cutadapt_pe:
         fastq2="results/trimmed/{sample}-{unit}.2.fastq.gz",
         qc="results/trimmed/{sample}-{unit}.qc.txt",
     params:
-        "{}".format(config["params"]["cutadapt-pe"]),
+        extra="{}".format(config["params"]["cutadapt-pe"]),
     log:
         "results/logs/cutadapt/{sample}-{unit}.log",
     wrapper:
-        "0.31.1/bio/cutadapt/pe"
+        "v1.22.0/bio/cutadapt/pe"
 
 
-rule cutadapt:
+if is_3prime_experiment:
+
+    # Rule cutadapt1 checks and removes poly-A tails and sequence qualilty score <20.
+    # reasoning behind parameters:
+    #   * `-m 20`:
+    #   * Discards any read under 20bp by -m 20
+    #   * `-O 20 -a ""polyA=A{20}"" -a ""QUALITY=G{20}"" -n 2`:
+    #   * Removes reads having polyA (18bp) or polyG (18bp) and repeat the process twice by -n 2
+    # In short, it removes reads that are predominantly polyA or polyG.
+
+    # Rule cutadapt2 checks if reads are obtained from nextseq and removes adapters specific to nextseq
+    # reasoning behind parameters:
+    #   * `-m 20`:
+    #   * Discards any read under 20bp by -m 20
+    #   * `--nextseq-trim=10`
+    #   * Quality score<20 is trimmed, based on NextSeq/NovaSeq 2-color SBS system
+    #   * `-a A{18}AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC;min_overlap=3;max_error_rate=0.100000`
+    #   * Trims the polyA-stretch + adapter at the 3' end with minimum overlap 3bp, maximum error of 1bp every 10bp
+
+    # Rule cutadapt3 removes final set of adapters from 5'end and discards reads based on trimmed read length
+    # reasoning behind parameters:
+    #   * `-m 20`:
+    #   * Discards any read under 20bp by -m 20
+    #   * `-O 20 -g r1adapter=AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC;min_overlap=20 --discard-trimmed"`
+    #   * Trim the 5' end of adapter with miminum overlap of 20bp and discard its read by --discard-trimmed
```

Comment:
THis comes from lexogene, right?

1. Very important: please cite the page/manual you have this from here.
2. Move these rules into trim_3prime.smk

Reply:
Noted.